### DISCUSSION TITLE
Swap arguments?

### DISCUSSION THREAD

**Discussion by 9mm** [nvim-mini/mini.nvim#618] [General]
This is probably i find myself doing most and right now it seems treesitter only works with JS, and even then its spotty at best, for example some destructured argument `function({ a, b })` will break.

And on top of that, it doesnt work in any other languages.

Do any of your plugins do this? I searched around and `ai` seemed closest but it didnt seem to allow movement.

Such a great series of plugins youre doing, thank you !! 🎩

**Comment by echasnovski** [comment]
No module addresses this directly, no.

What I use myself is a combination of next/prev [argument textobject](https://github.com/echasnovski/mini.nvim/blob/333d2d1090c80ac936b960469a6e93982cbaeb21/doc/mini-ai.txt#L163) from 'mini.ai' and [exchange operator](https://github.com/echasnovski/mini.nvim/blob/333d2d1090c80ac936b960469a6e93982cbaeb21/doc/mini-operators.txt#L231-L241) from 'mini.operators'. I have keymaps similar to this:

```lua
vim.keymap.set('n', '<Leader>oh', '<Cmd>normal gxiagxila<CR>')
vim.keymap.set('n', '<Leader>ol', '<Cmd>normal gxiagxina<CR>')
```

They are **far from perfect**,  as `ina`/`ila` textobjects not always mean what I would like it to mean (in case of some complex nested arguments), but in non-nested cases it works good.

**Comment by 9mm** [comment]
Super awesome, I'm gioing to pour some coffee and have a look. Even if its a little better than what im dealing with its going to be so amazing

**Comment by 9mm** [comment]
Dude this works so well... man your plugins are just the best. It has made my experience switching from vim to neovim 10/10

**Comment by 9mm** [comment]
mini.operators is also amazing, i can't believe I missed that one. Multiply + exchange is freakin sweet

**Comment by 9mm** [comment]
one easy Q: with `gs` how do I do case insensitive?

**Reply by echasnovski** [reply]
No easy solution, I am afraid, as by default it `sort()` operator just compares strings with `<` and Lua does this case sensitively.

You can [override compare function](https://github.com/echasnovski/mini.nvim/blob/333d2d1090c80ac936b960469a6e93982cbaeb21/doc/mini-operators.txt#L461-L462), though.

**Comment by 9mm** [comment]
Also one other easy `gs` question. Lets say I have this:

```javascript
    watch(accountId, () => {
      fetchData();
    });

    return {
      accountId,
      accounts,
      [c]olumns, <- cursor is here
      data,
      loading,
      query,
    };
  },
```

How do I sort everything in the `{`?

I have tried `gsii, gsi{ gsip, gsai` and i cant seem to nail it right. I suspect `gsi{` should work, but it makes a weird result:

```
    return {
      ,
      accountId,
      accounts,
      columns,
      data,
      loading,query
    };
```

If i do the same selection of `vii` and then `:sort i` it does work however

**Reply by echasnovski** [reply]
> I have tried `gsii, gsi{ gsip, gsai` and i cant seem to nail it right. I suspect `gsi{` should work, but it makes a weird result:

Try `gsVii` or `Viigs`. Default sort function [sorts charwise and linewise/blockwise regions differently](https://github.com/echasnovski/mini.nvim/blob/333d2d1090c80ac936b960469a6e93982cbaeb21/doc/mini-operators.txt#L435-L442).

**Reply by 9mm** [reply]
Awesome, I should have just read that, thank you!
